app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,48 @@
+require 'date'
+
+class SalesController < ApplicationController
+
+  def show
+    @sale = Sale.where(vendor_id: params[:id]).order(purchase_time: :asc)
+    @group = Sale.group("strftime('%m', purchase_time)").count
+    @current = Time.now.strftime('%m')
+    render :show
+  end
+
+  def month_sale
```

Comment:
There is a lot going on in this controller method. You should check whether or not you need all of these instance variables.  It might also be worthwhile to break this method down into several methods that each do one task.